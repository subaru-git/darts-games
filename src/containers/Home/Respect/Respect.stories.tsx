/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Respect from './Respect';

export default {
  title: 'Containers/Home/Respect',
  component: Respect,
} as ComponentMeta<typeof Respect>;

const Template: ComponentStory<typeof Respect> = (args) => <Respect {...args} />;

export const Default = Template.bind({});
Default.args = {};
