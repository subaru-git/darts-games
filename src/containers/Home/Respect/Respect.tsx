'use client';
import React, { FC } from 'react';
import LiteYoutubeEmbed from 'react-lite-youtube-embed';
import 'react-lite-youtube-embed/dist/LiteYouTubeEmbed.css';
import { useBreakpoint } from '@/hooks/tailwind';
import { useTranslations } from 'next-intl';

const Respect: FC = () => {
  const isMd = useBreakpoint('md');
  return <>{isMd ? <DesktopRespect /> : <MobileRespect />}</>;
};

const DesktopRespect: FC = () => {
  const t = useTranslations('Home.description');
  return (
    <div className='flex justify-center gap-8'>
      <div className='h-[360px] w-[640px]'>
        <LiteYoutubeEmbed
          id='HiNOjEYbUXc'
          title='【ダーツ】友達にダーツをさせる方法【菊地山口】'
          webp
        />
      </div>
      <div className='flex max-w-lg flex-col justify-center gap-4'>
        <h2 className='text-2xl font-bold'>{t('title')}</h2>
        <p className='whitespace-pre-wrap text-base'>{t('description')}</p>
      </div>
    </div>
  );
};

const MobileRespect: FC = () => {
  const t = useTranslations('Home.description');
  return (
    <div className='flex flex-col items-center gap-4'>
      <div className='h-[180px] w-[320px]'>
        <LiteYoutubeEmbed
          id='HiNOjEYbUXc'
          title='【ダーツ】友達にダーツをさせる方法【菊地山口】'
          webp
        />
      </div>
      <div className='flex flex-col gap-4'>
        <h2 className='text-center text-lg font-bold'>{t('title')}</h2>
        <p className='whitespace-pre-wrap text-sm'>{t('description')}</p>
      </div>
    </div>
  );
};

export default Respect;
