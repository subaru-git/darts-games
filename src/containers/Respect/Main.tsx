import React, { FC } from 'react';
import CardView from '@/containers/Respect/CardView';
import MainTemplate from '@/templates/MainTemplate';

type MainProps = {
  data: RespectResult;
};

const Main: FC<MainProps> = ({ data }) => {
  return (
    <MainTemplate label='respect-main'>
      <CardView data={data} />
    </MainTemplate>
  );
};

export default Main;
