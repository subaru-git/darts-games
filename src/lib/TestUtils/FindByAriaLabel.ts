export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
