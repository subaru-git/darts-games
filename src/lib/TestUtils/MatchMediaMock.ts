import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
