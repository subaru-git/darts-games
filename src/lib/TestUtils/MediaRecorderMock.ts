export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
