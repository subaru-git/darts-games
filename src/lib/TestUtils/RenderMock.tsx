import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';
import { NextIntlClientProvider } from 'next-intl';

export const render = (children: ReactNode | ReactNode[]) => {
  const messages = require('@/locales/en.json');
  return defaultRender(
    <ChakraProvider>
      <NextIntlClientProvider messages={messages} locale='en'>
        <Providers>{children}</Providers>
      </NextIntlClientProvider>
    </ChakraProvider>,
  );
};
