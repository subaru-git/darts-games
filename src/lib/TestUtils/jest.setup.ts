import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
