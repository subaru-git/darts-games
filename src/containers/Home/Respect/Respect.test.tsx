import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import Respect from './Respect';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
test('should rendering', () => {
  window.matchMedia = createMatchMedia(1100);
  const { container } = render(
    <ChakraProvider>
      <Respect />
    </ChakraProvider>,
  );
  expect(container.getElementsByClassName('YouTube')).not.toBeNull();
  expect(container).toMatchSnapshot();
});
