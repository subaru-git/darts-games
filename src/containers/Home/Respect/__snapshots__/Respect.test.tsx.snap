// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="flex justify-center gap-8"
  >
    <div
      class="h-[360px] w-[640px]"
    >
      <link
        as="image"
        href="https://i.ytimg.com/vi_webp/HiNOjEYbUXc/hqdefault.webp"
        rel="preload"
      />
      <article
        class="yt-lite "
        data-title="【ダーツ】友達にダーツをさせる方法【菊地山口】"
        style="background-image: url(https://i.ytimg.com/vi_webp/HiNOjEYbUXc/hqdefault.webp); --aspect-ratio: 56.25%;"
      >
        <button
          aria-label="Watch 【ダーツ】友達にダーツをさせる方法【菊地山口】"
          class="lty-playbtn"
          type="button"
        />
      </article>
    </div>
    <div
      class="flex max-w-lg flex-col justify-center gap-4"
    >
      <h2
        class="text-2xl font-bold"
      >
        Suggest practicing home darts
      </h2>
      <p
        class="whitespace-pre-wrap text-base"
      >
        We propose practice methods for home darts, which tend to be monotonous.
The practice is supported as an application with references to YouTube, Blogs, etc.
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
